### PR TITLE
[libc] implement getrandom for windows

### PR DESCRIPTION
**Pull request by SchrodingerZhu**
This PR implements the `getrandom` emulation using [`BCryptGenRandom`](https://learn.microsoft.com/en-us/windows/win32/api/bcrypt/nf-bcrypt-bcryptgenrandom).

### CODE REVIEW

File: libc/hdr/types/CMakeLists.txt [hunk 1]
```diff
@@ -93,6 +93,14 @@ add_proxy_header_library(
...
 )
 
+add_proxy_header_library(
+  ssize_t
+  HDRS
+    ssize_t.h
+  FULL_BUILD_DEPENDS
+    libc.include.llvm-libc-types.ssize_t
+)
+
```
Comment (michaelrj-google) [inline libc/hdr/types/CMakeLists.txt:103]: this is important, but should probably be in a separate patch.

File: libc/include/llvm-libc-types/ssize_t.h [hunk 2]
```diff
@@ -9,6 +9,12 @@
 #ifndef LLVM_LIBC_TYPES_SSIZE_T_H
 #define LLVM_LIBC_TYPES_SSIZE_T_H
 
+// https://learn.microsoft.com/en-us/windows/win32/winprog/windows-data-types
+#if __has_include(<BaseTsd.h>)
```
Comment (michaelrj-google) [inline libc/include/llvm-libc-types/ssize_t.h:13]: `__has_include` isn't the appropriate way to check if this is a windows target. I'd recommend doing `#ifdef WIN32` instead.

**Review by michaelrj-google — COMMENTED** [review]
Is this necessary? `getrandom` is a linux specific function.

**Comment by SchrodingerZhu** [comment]
Good point. I made this there are other parts of code that assumes getrandom on default. Maybe I should chane them instead. Closing this now. I will separate out the ssize_t changes.
